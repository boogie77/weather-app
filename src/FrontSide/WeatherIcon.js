import React from "react";

export const WeatherIcon = ({ icon, onClick }) => {
  return (
    <img
      src={`/icons/${icon}.svg`}
      width="120px"
      alt="weather icon"
      onClick={onClick}
    />
  );
};
