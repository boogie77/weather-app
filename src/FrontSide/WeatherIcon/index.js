import React from "react";
import PropTypes from "prop-types";

const WeatherIcon = ({ icon, onClick }) => {
  return (
    <img
      src={`/icons/${icon}.svg`}
      width="120px"
      alt="weather icon"
      onClick={onClick}
    />
  );
};

WeatherIcon.propTypes = {
  icon: PropTypes.string,
  onClick: PropTypes.func
};

export default WeatherIcon;
