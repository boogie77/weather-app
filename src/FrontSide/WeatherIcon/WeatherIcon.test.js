import React from "react";
import WeatherIcon from "./index";
import toJson from "enzyme-to-json";
import { shallow } from "enzyme";

describe("<WeatherIcon />", () => {
  it("renders and matches snapshot", () => {
    const wrapper = shallow(<WeatherIcon icon="testIcon" />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
