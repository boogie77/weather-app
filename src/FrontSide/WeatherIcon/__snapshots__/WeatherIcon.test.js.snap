// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<WeatherIcon /> renders and matches snapshot 1`] = `
<img
  alt="weather icon"
  src="/icons/testIcon.svg"
  width="120px"
/>
`;
